src/components/checkbox/checkbox.spec.ts

```diff
@@ -0,0 +1,713 @@
+        let el = fixture.debugElement.query(By.css('.md-checkbox'));
+        expect(el.nativeElement.getAttribute('role')).toEqual('checkbox');
+      }).then(done).catch(done);
+    });
+
+    it('defaults "aria-checked" to "false" on the host element', function(done: () => void) {
+      builder.createAsync(CheckboxController).then(function(fixture) {
+        fixture.detectChanges();
+        let el = fixture.debugElement.query(By.css('.md-checkbox'));
+        expect(el.nativeElement.getAttribute('aria-checked')).toEqual('false');
+      }).then(done).catch(done);
+    });
+
+    it('defaults "aria-disabled" to "false" on the host element', function(done: () => void) {
+      builder.createAsync(CheckboxController).then(function(fixture) {
+        fixture.detectChanges();
+        let el = fixture.debugElement.query(By.css('.md-checkbox'));
+        expect(el.nativeElement.getAttribute('aria-disabled')).toEqual('false');
+      }).then(done).catch(done);
+    });
+
+    it('defaults tabindex to 0 to the host element', function(done: () => void) {
+      builder.createAsync(CheckboxController).then(function(fixture) {
+        fixture.detectChanges();
+        let el = fixture.debugElement.query(By.css('.md-checkbox'));
+        expect(el.nativeElement.tabIndex).toEqual(0);
+      }).then(done).catch(done);
+    });
+
+    it('allows clients to provide their own tabindex', function(done: () => void) {
+      builder.createAsync(CheckboxCustomTabindexController).then(function(fixture) {
+        fixture.detectChanges();
+        let component = fixture.componentInstance;
+        let el = fixture.debugElement.query(By.css('.md-checkbox'));
+        expect(el.nativeElement.tabIndex).toEqual(component.checkboxTabindex);
+      }).then(done).catch(done);
+    });
+
+    it('allows clients to provide an aria-label', function(done: () => void) {
+      builder.createAsync(CheckboxCustomArialabelController).then(function(fixture) {
+        fixture.detectChanges();
+        let component = fixture.componentInstance;
+        let el = fixture.debugElement.query(By.css('.md-checkbox'));
+        expect(el.nativeElement.getAttribute('aria-label')).toEqual(component.checkboxLabel);
+      }).then(done).catch(done);
+    });
+
+    it('sets the "aria-labelledby" attribute to the id of the label', function(done: () => void) {
+      builder.createAsync(CheckboxController).then(function(fixture) {
+        fixture.detectChanges();
+        let el = fixture.debugElement.query(By.css('.mc-checkbox'));
+        let label = el.nativeElement.querySelector('label');
+        expect(el.nativeElement.getAttribute('aria-labelledby')).toEqual(label.id);
+      }).then(done).catch(done);
+    });
+
+    describe('when given an "align" input with a value of "end"', function() {
+      var fixture: ComponentFixture;
+
+      beforeEach(function(done: () => void) {
+        builder.createAsync(CheckboxEndAlignedController).then(function(_fixture_) {
```

Comment:
Rather than `_fixture_`, I'd be fine with just saying `f` (since the first thing you do is set it)